group_project_v2/mixins.py

```diff
@@ -125,7 +126,180 @@ def real_user_id(self, anonymous_student_id):
         return self._known_real_user_ids[anonymous_student_id]
 
 
-class WorkgroupAwareXBlockMixin(UserAwareXBlockMixin, CourseAwareXBlockMixin):
+class SettingsMixin(object):
+
+    def _get_setting(self, setting, default):
+        result = default
+        settings_service = self.runtime.service(self, "settings")
+        if settings_service:
+            xblock_settings = settings_service.get_settings_bucket(self)
+            if xblock_settings and setting in xblock_settings:
+                result = xblock_settings[setting]
+        return result
+
+
+class AuthXBlockMixin(SettingsMixin, ProjectAPIXBlockMixin):
+
+    _DEFAULT_TA_ROLE = ('assistant',)
+
+    _SEE_DASHBOARD_ROLE_PERMS = "see_dashboard_role_perms"
```

Comment:
@jbzdak I believe it's either role or permission. Could you please rename it to `SEE_DASHBOARD_ROLE = "see_dashboard_role"`. Also, it might make sense to actually use "can_access", not "see", but up to you.